src/Aspire.Hosting.Azure.AIFoundry/AzureAIFoundryLocalResourceExtensions.cs

```diff
@@ -0,0 +1,259 @@
+
+            deployment.Parent = resource; // Ensure the deployment has the correct parent reference
+        }
+
+        var healthCheckKey = $"{resource.Name}_check";
+        resourceBuilder.ApplicationBuilder.Services.AddHealthChecks()
+                .Add(new HealthCheckRegistration(
+                    healthCheckKey,
+                    sp => new FoundryHealthCheck(sp.GetRequiredService<FoundryLocalManager>()),
+                    failureStatus: default,
+                    tags: default,
+                    timeout: default
+                    ));
+
+        resourceBuilder.WithHealthCheck(healthCheckKey);
+
+        return resourceBuilder;
+    }
+
+    private static IResourceBuilder<AzureAIFoundryResource> WithInitializer(this IResourceBuilder<AzureAIFoundryResource> builder)
+    {
+        builder.ApplicationBuilder.Eventing.Subscribe<InitializeResourceEvent>(builder.Resource, (@event, ct)
+            => Task.Run(async () =>
+            {
+                var resource = (AzureAIFoundryResource)@event.Resource;
+                var rns = @event.Services.GetRequiredService<ResourceNotificationService>();
+                var manager = @event.Services.GetRequiredService<FoundryLocalManager>();
+
+                await rns.PublishUpdateAsync(resource, state => state with
+                {
+                    State = new ResourceStateSnapshot(KnownResourceStates.Starting, KnownResourceStateStyles.Info)
+                }).ConfigureAwait(false);
+
+                await manager.StartServiceAsync(ct).ConfigureAwait(false);
+
+                resource.ApiKey = manager.ApiKey;
+
+                if (manager.IsServiceRunning)
+                {
+                    if (resource.TryGetLastAnnotation<EndpointAnnotation>(out var endpoint))
+                    {
+                        endpoint.AllocatedEndpoint = new AllocatedEndpoint(
+                            new EndpointAnnotation(
+                                System.Net.Sockets.ProtocolType.Tcp,
+                                manager.Endpoint.Scheme,
+                                "http",
+                                "http",
+                                manager.Endpoint.Port,
+                                manager.Endpoint.Port,
+                                false,
+                                false),
+                            manager.Endpoint.ToString(),
+                            manager.Endpoint.Port);
+                    }
+
+                    await rns.PublishUpdateAsync(resource, state => state with
+                    {
+                        State = KnownResourceStates.Running,
+                        Properties = [.. state.Properties, new(CustomResourceKnownProperties.Source, "Foundry Local")]
+                    }).ConfigureAwait(false);
+                }
+                else
+                {
+                    await rns.PublishUpdateAsync(resource, state => state with
```

Comment:
No way to restart? No way to see logs as to why it's not starting...

Reply:
From my testing, there's no way to push logs from a "background task" such as this up to the parent resource log stream. But also, the SDK we use to manage Foundry Local allow us to hook into that anyway, as it manages the calls to the CLI or control plane (depending on the call you're making), so a failure is opaque to us.

As for the ability to restart/etc. let's tackle that via a future PR - this one can focus on MVP of Foundry Local + Azure AI Foundry, then add the nice-to-haves.